mistletoe/block_token.py

```diff
@@ -744,8 +776,8 @@ def __init__(self, content, align=None):
 
 class Footnote(BlockToken):
     """
-    Footnote token. A "link reference definition" according to the spec.
-    This is a leaf block token. Its children are inline (span) tokens.
+    Footnote token. A sequence of "link reference definitions" according to the spec.
```

Comment:
This is tricky. What I do know is that:
- A single Footnote.read() call will parse as many link reference definitions as it can before stopping.
- The Footnote itself won't store any information; it goes into _root_token.footnotes.
- The swap-in replacement I made for Footnote is a LinkReferenceDefinitionBlock (a block token), which in turn contains LinkReferenceDefinitions (a span token) as children. This was the design option that matched the rest of the codebase most closely, and makes both parsing and rendering most straightforward.
- The second removed line was misleading because Footnote tokens do not have children.

The reason for modifying the comment is mainly the first point here: that it will consume all link reference definitions as it can when parsing. All in all I think the updated description is more accurate, so I'd suggest keeping it. Or maybe rewriting it again to something better. Any suggestions?

Reply:
I've updated this, but I'm not 100% it ended up the way you intended, so please take another look.